### ISSUE TITLE
One of parameters (value: ) is not of the type Integer

### ISSUE THREAD

**Issue by adam-devapp** [progressivetech/net.ourpowerbase.remoteform#48]
When trying to manage a contribution page, the following error is shown:

```
 #cause: null
    -_trace: null
    #message: "One of parameters  (value: ) is not of the type Integer"
    #code: 0
    #file: "/home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Utils/Type.php"
    #line: 470
    trace: {
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Utils/Type.php:470 {
        ›   $data = htmlentities($data);
        ›   throw new CRM_Core_Exception("$name (value: $data) is not of the type $type");
        › }
      }
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Core/DAO.php:1721 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Core/DAO.php:1603 { …}
      /home/xxxx/xxxx/wp-content/uploads/civicrm/ext/net.ourpowerbase.remoteform/remoteform.php:360 { …}
      /home/xxxx/xxxx/wp-content/uploads/civicrm/ext/net.ourpowerbase.remoteform/remoteform.php:188 { …}
      /home/xxxx/xxxx/wp-content/uploads/civicrm/ext/net.ourpowerbase.remoteform/remoteform.php:290 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Utils/Hook.php:275 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Utils/Hook/WordPress.php:136 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/Civi/Core/CiviEventDispatcher.php:241 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/vendor/symfony/event-dispatcher/EventDispatcher.php:214 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/vendor/symfony/event-dispatcher/EventDispatcher.php:44 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/Civi/Core/CiviEventDispatcher.php:198 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Utils/Hook.php:167 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Utils/Hook.php:471 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Core/Form.php:662 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Core/QuickForm/Action/Display.php:76 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/packages/HTML/QuickForm/Controller.php:203 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/packages/HTML/QuickForm/Page.php:103 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Core/Controller.php:352 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Utils/Wrapper.php:98 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Core/Invoke.php:286 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Core/Invoke.php:69 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm/CRM/Core/Invoke.php:36 { …}
      /home/xxxx/xxxx/wp-content/plugins/civicrm/civicrm.php:1178 { …}
      /home/xxxx/xxxx/wp-includes/class-wp-hook.php:303 { …}
      /home/xxxx/xxxx/wp-includes/class-wp-hook.php:327 { …}
      /home/xxxx/xxxx/wp-includes/plugin.php:470 { …}
      /home/xxxx/xxxx/wp-admin/admin.php:259 { …}
      ```

**Comment by adam-devapp** [comment]
The payment gateway returns as an array, not an int:
array(2) { [0]=> string(1) "1" [1]=> string(1) "5" }

Hence why it fails.

**Comment by jmcclelland** [comment]
What version of CiviCRM are you using?

Judging from your back trace, the problem is with the call to `civicrm_api3('ContributionPage', 'get', $cp_params);`. We're asking for the `payment_processor' field and expecting just an id, but you are getting an array.

This smells like behavior that may have changed during a core upgrade. I'm just not sure if you are experience the old behavior or the new behavior. 'm running 5.39 and I get just an ID from `payment_processor` - if you are running a newer version then this sounds like a bug. If you are running an older version, then you'll probably need to upgrade CiviCRM.

**Comment by mlutfy** [comment]
I have to admit that we stumbled on this error, but I just assumed it was specific to our setup, and we don't use remoteform for contribution pages, so we just patch it out:

https://github.com/progressivetech/net.ourpowerbase.remoteform/blob/master/remoteform.php#L188

I think it happens when 2 or more processors are enabled?

**Comment by adam-devapp** [comment]
Yes Paypal and a credit card are configured.

Running CiviCRM 5.43.2 (latest).

**Comment by adam-devapp** [comment]
Can't disable on that site, as it is a live site. The variable is expecting an int, but array returned with multiple.. So potentially a new condition to handle an array could work?

**Comment by jmcclelland** [comment]
I'm afraid the particular bug you are experiencing is just the tip of the iceberg. We could fix the code to test for an array, but if it's an array, then what? The remote form UI expects one payment processor. Also, I've never tested with Pay Pal and, especially if it's PayPal standard that expects a redirect to the Pay Pal site, I am not confident it would work.

I typically creat dedicated Contrbution Pages to be used with Remote Form, since it only supports a subset of Contribution Page features. I'd suggst creating a new Contribution Page and experiment with using Remote Form on the brand new page, so you can make adjustments and tests without affecting your live page.

Unfortunately, I suspect this won't work with Pay Pal. And if it does, I suspect it won't work with multiple payment processors even if we fix the immediate code problem. 

I would support a pull request to enable that feature, but unfortunately don't have time to work on it myself.

**Comment by mlutfy** [comment]
How about displaying a message that remoteform is not supported on forms that have 2+ processors? (and only supports Stripe or pay-later)

**Comment by adam-devapp** [comment]
The client isn't actually using the contribution form with remote form - They are using a profile to post contact details for a newsletter signup.

**Comment by jmcclelland** [comment]
Is Remove Form enabled on any of your contribution pages. I'm not sure why it would trigger an error on a profile for a contribution page with more then one payment processor (although that sounds kinda like what mlufty is saying as well).

**Comment by jmcclelland** [comment]
Ah. I finally understand. The problem is with displaying the manage contribution page, not with remote form properly working. I see that clearly in the original message but somehow it only now has sunk in.

I just posted a first pass at a fix: 848edf4c292f1784e7554bdeca942fb47cb622ab

Ideally I would like to disable the remote form option if a contribution page has two or more processors enabled. But getting the logic to work properly is more then I can muster at the moment so I'm setting on a simple message.

I think this issue is now resolved, but feel free to re-open if there are still problems.
